qiskit/circuit/__init__.py

```diff
@@ -54,81 +78,825 @@
+``QuantumRegister("my_qreg", 5)``.  This has the advantage that you can give that collection of bits
+a name, which will appear during circuit visualizations (:meth:`QuantumCircuit.draw`) and exports to
+interchange languages (see :mod:`.qasm2` and :mod:`.qasm3`).  You can also pass a name and a list of
+pre-constructed bits, but this creates an "aliasing register", which are very poorly supported on
+hardware.
+
+Circuits track registers, but registers themselves impart almost no behavioral differences on
+circuits.  The only exception is that :class:`ClassicalRegister`\ s can be implicitly cast to
+unsigned integers for use in conditional comparisons of :ref:`control flow operations
+<circuit-control-flow>`.
+
+Classical registers and bits were the original way of representing classical data in Qiskit, and
+remain the most supported currently.  Longer term, the data model is moving towards a more complete
+and strongly typed representation of a range of classical data (see
+:ref:`circuit-repr-near-time-classical`), but you will still very commonly use classical bits in
+current Qiskit.
+
+
+Instruction contexts
+--------------------
+
+The scalar type of the :attr:`QuantumCircuit.data` sequence is the "instruction context" object,
+:class:`CircuitInstruction`.  This is essentially just a data class that contains a representation
+of what is to be done (its :attr:`~CircuitInstruction.operation`), and the data it acts on (the
+:attr:`~CircuitInstruction.qubits` and :attr:`~CircuitInstruction.clbits`).
+
+.. autoclass:: CircuitInstruction
+    :members:
+
+Programmatically, this class is actually implemented in Rust and is a constructed handle to internal
+data within Rust space.  Mutations to instances of this class will not be reflected in the circuit.
+In general, you cannot mutate instruction contexts that are already in the circuit directly; the
+:class:`QuantumCircuit` interface is designed for storing and building circuits, while the
+:ref:`transpiler and its passes <qiskit-transpiler>`, and its intermediate :class:`.DAGCircuit`
+representation, are where you should look for an interface to mutate circuits.
+
+The :class:`QuantumCircuit` methods that add instructions to circuits (such as
+:meth:`~QuantumCircuit.append`, and all the helper standard-gate methods) return an
+:class:`InstructionSet`, which is a handle to several :class:`CircuitInstruction`\ s simultaneously.
+
+.. autoclass:: InstructionSet
+    :members:
+
+This :class:`InstructionSet` is now little used in Qiskit.  It provides a very minimal set of
+methods to perform post-append mutations on instructions (which *will* be propagated to the
+circuit), but these are now discouraged and you should use the alternatives noted in those methods.
+
+
+Operations, instructions and gates
+----------------------------------
+
+Within a :class:`CircuitInstruction`, the minimal interface that any operation must fulfil is
```

Comment:
"fulfil" looks strange but apparently this is british english :-)

Reply:
Oh yeah, if you spot any Britishism feel free to offer the Americanisations as suggestions - I _try_ and write with American spelling for these things, but I'm fighting 30 years of conditioning haha.

Reply:
Fixed in 2c1cc17232a.